app/models/profile.rb

```diff
@@ -100,30 +100,18 @@ def from_omniauth_hash(omniauth_user_hash)
   end
 
   def image_url= url
-    return image_url if url == ''
-    if url.nil? || url.match(/^https?:\/\//)
-      super(url)
-    else
-      super(absolutify_local_url(url))
-    end
+    return image_url if url == '' or url.nil?
+    super(url[/(https?:\/\/(.(?!\/))*.)?(.*)/, 3])
   end
 
   def image_url_small= url
-    return image_url if url == ''
-    if url.nil? || url.match(/^https?:\/\//)
-      super(url)
-    else
-      super(absolutify_local_url(url))
-    end
+    return image_url if url == '' or url.nil?
+    super(url[/(https?:\/\/(.(?!\/))*.)?(.*)/, 3])
   end
 
   def image_url_medium= url
-    return image_url if url == ''
-    if url.nil? || url.match(/^https?:\/\//)
-      super(url)
-    else
-      super(absolutify_local_url(url))
-    end
+    return image_url if url == '' or url.nil?
+    super(url[/(https?:\/\/(.(?!\/))*.)?(.*)/, 3])
```

Comment:
Let's take this opportunity and DRY up that code by extracting it into a method. Instead of using `super`, you can use the fields name as a symbol and use the `[]=` method to set the value.